Bump to version 0.12.0

FintanH opened the pull request:
Currently includes #69

greyblake committed:
fix: use vec! in nonempty!

As pointed out[^0], `nonempty!` runs into macro hygiene issues. Use the same
approach as tracing[^1] to re-export the `vec!` macro and qualify it using the
`$crate` syntax.

[^0]: https://github.com/cloudhead/nonempty/pull/70#discussion_r2008824990
[^1]: https://github.com/tokio-rs/tracing/blob/e63ef57f3d686abe3727ddd586eb9af73d6715b7/tracing/src/lib.rs#L987

FintanH committed:
Bump to version 0.12.0

Signed-off-by: Fintan Halpenny <fintan.halpenny@gmail.com>
X-Clacks-Overhead: GNU Terry Pratchett

teohhanhui commented on src/lib.rs:
```diff
@@ -134,7 +134,7 @@ macro_rules! nonempty {
         $crate::NonEmpty {
             head: $h,
-            tail: alloc::vec::Vec::new(),
+            tail: vec![],
```
This runs into issues with macro hygiene:
* https://doc.rust-lang.org/reference/macros-by-example.html#hygiene
* https://veykril.github.io/tlborm/decl-macros/minutiae/hygiene.html

It would have to be something like `$crate::__macro_support::vec` instead.

See e.g. https://github.com/tokio-rs/tracing/pull/1918

epatters commented:
Would it be possible to merge this soon? Getting a fix to #68 would be much appreciated!

FintanH committed:
feat: update bincode optional feature

It was a misnomer to name the feature flag `bincode2` since that is the name of
unmaintained library. The library `bincode` is actively maintained, so the
feature flag is named after it instead.

There is a proper version release v2.0.1 now, which is specified instead. The
derive is cleaned up, and there is a roundtrip test added.

FintanH committed:
chore: link `FromIterator` and `Iterator` docs

FintanH committed:
fix: NonEmpty::capacity returns NonZeroUsize

FintanH committed:
chore: update github action versions
